client/src/components/Shop/shop.js

```diff
@@ -0,0 +1,270 @@
+  }
+
+  render() {
+    let modalClose = () => this.setState({ modalShow: false });
+
+    return (
+      <div className="container">
+        <MyVerticallyCenteredModal
+          show={this.state.modalShow}
+          text="example text goes here"
+          onHide={modalClose}
+        />
+        <div className="row form-group">
+          <div className="col-sm-10">
+            <h4>
+              {this.props.name}: ${this.props.price}
+            </h4>
+          </div>
+          <div className="col-sm-2 text-right">qty: {this.state.qty}</div>
+        </div>
+        <div className="row btn-toolbar">
+          <div className="col-6">
+            <button
+              className="waves-effect waves-light btn btn-outline-primary"
+              onClick={this.showInfo}
+            >
+              show info
+            </button>
+          </div>
+          <div className="col-6 text-right">
+            <button
+              className="waves-effect waves-light btn btn-outline-primary"
+              onClick={this.add}
+            >
+              +1
+            </button>
+            <button
+              className="waves-effect waves-light btn btn-outline-primary"
+              onClick={this.subtract}
+              disabled={this.state.qty < 1}
+            >
+              -1
+            </button>
+          </div>
+        </div>
+        <hr />
+        <Button
+          variant="primary"
+          onClick={() => this.setState({ modalShow: true })}
+        >
+          Launch modal
+        </Button>
+      </div>
+    );
+  }
+}
+
+/* Total */
+class Total extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    let total = this.props.total.toFixed(2);
+
+    return (
+      <div className="container">
+      <div
+        style={{
+          marginTop: "30px",
+          backgroundColor: "#F6F6F6",
+          padding: "10px"
+        }}
+      >
+        <h3 className="row" style={{ fontWeight: 400 }}>
+          <span className="col-6">total price:</span>
+          <span className="col-6 text-right">${total}</span>
+        </h3>
+      </div>
+      </div>
+    );
+  }
+}
+// TODO: Create CHECKOUT btn to debit from piggy bank
+
+/* ProductForm */
+class ProductForm extends Component {
+  submit(e) {
+    e.preventDefault();
+    var product = {
+      name: this.refs.name.value,
+      price: Number(this.refs.price.value),
+      info: this.refs.info.value
+    };
+    console.log(this.refs.name.value, this.refs.price.value);
+    this.props.handleProduct(product);
+    this.refs.name.value = "";
+    this.refs.price.value = 0;
+    this.refs.info.value = "";
+  }
+
+  render() {
+    return (
+      <div className="container">
+      <form onSubmit={this.submit.bind(this)}>
+        <h3>add item</h3>
+        <div className="row form-group">
+          <label className="col-sm-2  col-sm-form-label">item name:</label>
+          <div className="col-sm-10">
```

Comment:
looks like there are repetitive inputs. We can extract this to it's own component and repeat it 3 times to condense the lines of code.